### PR TITLE
Fixes #18 - Make app drawer close when layout toggles wide

### LINKED ISSUE #18: menu drawer does not auto close if reverting to tabs
make the view port narrow, open the menu, widen the view port, you now have menu + tabs until you close the menu.

### PR DESCRIPTION
**Pull request by ktiedt**

### CODE REVIEW

File: patterns/transform-navigation/x-app.html [hunk 1]
```diff
@@ -109,8 +109,17 @@
...
 
-      }
+      },
+
+      observers: ['onLayoutChange(wideLayout)'],
+
+      onLayoutChange: function(wide) {
+        var drawer = this.$.drawer;
 
+        if (wide && drawer.opened) {
```
Comment (blasten) [inline patterns/transform-navigation/x-app.html]: It seems like this code is missing the the case for `!wide && !drawer.opened` is that case needed?

File: patterns/transform-navigation/x-app.html [hunk 2]
```diff
@@ -109,8 +109,17 @@
...
+          drawer.toggleDrawer();
```
Comment (blasten) [inline patterns/transform-navigation/x-app.html]: it would be a little more clear to use `drawer.opened = false;`

**Comment by blasten** [comment]
@ktiedt I left some comments. cc @frankiefu

File: patterns/transform-navigation/x-app.html (hunk 1 above)
Comment (ktiedt) [inline patterns/transform-navigation/x-app.html]: (responding from memory on this)

If the drawer is not open, and layout is !wide the drawer is opened via the menu icon which would negate the need for this check, I believe.

File: patterns/transform-navigation/x-app.html (hunk 2 above)
Comment (ktiedt) [inline patterns/transform-navigation/x-app.html]: More concise maybe, I can certainly make that change.

File: patterns/transform-navigation/x-app.html (hunk 1 above)
Comment (blasten) [inline patterns/transform-navigation/x-app.html]: :+1: I will check. thanks for the explanation!

**Comment by frankiefu** [comment]
@ktiedt Sorry for the late response.  Thanks for the PR and the solution looks fine.  Since we have now moved the `opened` property to `app-drawer`, could you update PR for that?  Thanks.

**Comment by ktiedt** [comment]
Updated, now I just need to squash things because this just got epically noisy

**Comment by ktiedt** [comment]
So, I revisited this with a proper branched pull request -- I mucked this one up something fierce, and I didn't even try very hard :/ #56 is the new PR.

**Pull request closed without merging**